modules/programs/piston-cli.nix

```diff
@@ -0,0 +1,50 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.piston-cli;
+  yamlFormat = pkgs.formats.yaml { };
+in {
+  meta.maintainers = with maintainers; [ ethancedwards8 ];
+
+  options.programs.piston-cli = {
+    enable = mkEnableOption "piston-cli, code runner";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.piston-cli;
+      defaultText = literalExample "pkgs.piston-cli";
+      description = "The piston-cli package to use.";
+    };
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = literalExample ''
+        {
+          # Code highlihting theme
+          theme = "emacs";
+          # Used for displaying your resultant output you get from the api
+          box_style = "MINIMAL_DOUBLE_HEAD";
+          # This is what is used for continuing your prompt when you go a new line
+          prompt_continuation = "...";
+          # This is the start symbol used when you are shell
+          prompt_start = ">>>";
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/piston-cli/config.yml</filename>.
```

Comment:
Can you provide a link to the documentation as well?
If the documentation is only available in a manpage, you can reference it with the following syntax:
```
<citerefentry>
  <refentrytitle>command</refentrytitle>
  <manvolnum>1</manvolnum>
</citerefentry>
```
where command is the name of man page and 1 the type of pages.

Reply:
Sadly the project has literally zero documentation. You have to sort through the code to find options. I listed all the possible (Yes, only four) in the example.

Reply:
Feel free to add a link to the documentation once the upstream project gets one.